CondFormats/JetMETObjects/src/SimpleJetCorrector.cc

```diff
@@ -78,17 +79,25 @@ float SimpleJetCorrector::correctionBin(unsigned fBin, const std::vector<float>&
     handleError("SimpleJetCorrector", sserr.str());
   }
   unsigned N = fY.size();
+
   if (N > 4) {
     std::stringstream sserr;
     sserr << "two many variables: " << N << " maximum is 4";
     handleError("SimpleJetCorrector", sserr.str());
   }
   const std::vector<float>& par = mParameters.record(fBin).parameters();
 
-  double params[par.size() - 2 * N];
+  auto nParams = std::max<int>(static_cast<int>(par.size()) - 2 * N, 1);
+  double params[nParams];
   for (unsigned int i = 2 * N; i < par.size(); i++) {
     params[i - 2 * N] = par[i];
   }
+  if (nParams < 1) {
+    std::stringstream sserr;
+    sserr << "nParams should be greater than 0";
+    handleError("SimpleJetCorrector", sserr.str());
+  }
+
   double x[4] = {};
```

Comment:
@laurenhay , should this be change to `double x[N] = {};` ?

Reply:
Yes I think this was previously x[4] bc of the check on line 83 that ensures the max length of 4, but there's no reason to allocate that memory if N is less than 4. I can make this change as well and add a check for N > 0.

Reply:
> should this be `double x[N] = {};`

I just want to note that variable-length arrays (`N` is a runtime variable) are not part of C++ standard (even if they are part of C99). GCC (and clang) implement them as an extension for C++ as well though, but I think it would be best to avoid them as non-standard.

> Yes I think this was previously x[4] bc of the check on line 83 that ensures the max length of 4

Does it? It has a check that calls `handleError()`, but as far as I can tell, that function only logs an error message, and this line is reached even in that case
https://github.com/cms-sw/cmssw/blob/fbfd2d7760e1f0828dce6d95bbaf81c73981ee8c/CondFormats/JetMETObjects/interface/Utilities.h#L110-L118

Reply:
@makortel Yes, it is a bit of a mess, but it's been working for 15 years. In reality this entire code base should be re-written since it's full of these little "foibles", but I think it's best to at least get rid of the undefined behavior first.

Reply:
Sorry yes I had realized that before about the error handling only being a log but forgot. I believe the max of 4 is required for invert() which uses another array x[4]. So maybe it's best for now to leave x[4] but can require N<=4 and !fY.empty().